app/javascript/components/remove-generic-item-modal.jsx

```diff
@@ -212,8 +213,6 @@ class RemoveGenericItemModal extends React.Component {
 
 RemoveGenericItemModal.propTypes = {
   dispatch: PropTypes.func.isRequired,
-  recordId: PropTypes.objectOf(PropTypes.any).isRequired,
```

Comment:
If  recordId and gridChecks are in props, then you should have them here.
1) remove `isRequired` for these two and add default props.
2) looks like gridChecks type is not object, that should be corrected.

in any case, we shouldn't be deleting the props here.